In [5]:
from google.colab import files
uploaded = files.upload()

Saving dataset_limpo_covid19_manaus.csv to dataset_limpo_covid19_manaus.csv


Visando efetuar uma análise exploratória dos dados, as seguintes questões foram respondidas: 

1. Quantos exemplos e atributos há na base de dados após a limpeza e organização?

2. Qual a porcentagem de indivíduos recuperados em relação ao todo?

3. Os casos acometeram mais indivíduos do sexo masculino ou feminino?

4. Qual a média e desvio padrão de idade dos indivíduos que contraíram a COVID-19? Qual o
indivíduo mais jovem e o mais idoso a contraírem tal enfermidade?

5. Qual o bairro com maior incidência de casos?

6. Quais os três bairros com maior incidência de casos recuperados?

7. Quais os tipos de testes efetuados, segundo os dados? Indicando os dados de maneira quantitativa
e percentual.

8. Qual taxa de letalidade pode ser calculada a partir do conjunto de dados? Para o cálculo desta
taxa foi considerada a fração do total de óbitos pelo total de casos;

9. Qual o tipo de correlação, mediante coeficiente de correlação de Pearson, entre a idade e o
número de casos? 


# Importando as biblioteca necessárias

In [6]:
import io
import pandas as pd



Para resposder tais questões, efetou-se a leitura do dataset após a limpeza dos dados.

In [7]:
# Dataset é armazenado em um Dataframe pandas
df = pd.read_csv(io.BytesIO(uploaded['dataset_limpo_covid19_manaus.csv']))

Uma rápida visualização do estado atual do conjunto de dados

In [8]:
df.head()

idade sexo  ... dt_notificacao                tipo_teste
0     25    F  ...     20/03/2020                    RT-PCR
1     41    F  ...     27/03/2020                    RT-PCR
2     37    M  ...     27/04/2020  TESTE RÁPIDO - ANTICORPO
3     44    M  ...     27/04/2020  TESTE RÁPIDO - ANTICORPO
4     27    F  ...     24/04/2020  TESTE RÁPIDO - ANTICORPO

[5 rows x 6 columns]

**1. Quantos exemplos e atributos há na base de dados após a limpeza e organização?**

Os nomes das colunas que sobraram após a limpeza dos dados, são os atributos.

In [ ]:
df.columns

Index(['idade', 'sexo', 'bairro', 'conclusao', 'dt_notificacao', 'tipo_teste'], dtype='object')

A quantidade de atributos pode ser obtida contando as colunas que sobraram.

In [ ]:
len(df.columns)

6

Para os exemplos basta fazer o processo análogo para a linhas do dataset, dessa vez, o comprimento do dataset já é a própria resposta.

In [ ]:
len(df)

6360

Portanto, após a limpeza dos dados temos 6 atributos e 6360 exemplos no dataset.

# Abaixo as contribuções usando Jupyter Notebook do membro HiagoOliver

### Tendo em vista um visão mais ampla do conjunto de dados contendo somente os casos confirmados realiza-se as seguintes perguntas:

#### 1. Quantos atributos descrevem cada exemplo? Quais são eles?

In [9]:
import pandas as pd

df = pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_limpo_covid19_manaus.csv")

print('%s atributos descrevem cada exemplo\n' % df.columns.size)
print('Sendo os atributos:\n')

for atributo in df.columns.values:
    print(atributo[1:])

6 atributos descrevem cada exemplo

Sendo os atributos:

dade
exo
airro
onclusao
t_notificacao
ipo_teste


### A coluna classificacao foi removida do dataset quando executada a limpeza sendo necessário utilizar o dataset original

#### 2. Quantos casos confirmados há em Manaus, cumulativamente?

##### Abordagem 1: Utilizando groupby() para verificação de casos confirmados.

In [15]:
import pandas as pd

df = pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_covid19_manaus.csv", encoding="ISO-8859-1", sep=";", dtype='unicode')

casos_distintos = df.groupby(by='_classificacao').size()

#print('Situação dos casos %s' % casos_distintos)

print('Em Manaus há {0} casos confirmados'.format(casos_distintos['Confirmado']))

Em Manaus há 36671 casos confirmados


#### Abordagem 2: Utilizando query() e count() para verificação de casos confirmados.

In [18]:
import pandas as pd

df = pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_covid19_manaus.csv", encoding="ISO-8859-1", sep=";")

consulta = df.query('_classificacao == "Confirmado"')['_classificacao'].count()
print('Em Manaus há %s casos confirmados' % consulta)

Em Manaus há 36671 casos confirmados


#### Abordagem 3: Utilizando size para verificação de casos confirmados.

In [24]:
import pandas as pd

df= pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_covid19_manaus.csv", encoding="ISO-8859-1", sep=";")

df = df[df['_classificacao'] == "Confirmado"]
print('Em Manaus há %s casos confirmados' % df['_classificacao'].size)


Em Manaus há 36671 casos confirmados


#### Abordagem 4: Verificando todos as classificações de casos

In [32]:
import pandas as pd

df = pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_covid19_manaus.csv", encoding="ISO-8859-1", sep=";")

casos = df['_classificacao'].unique()

for caso in casos:
    data = df[df['_classificacao'] == caso]
    print('Em Manaus há {1} casos classificados como {0}'.format(caso, data['_classificacao'].size))

Em Manaus há 36671 casos classificados como Confirmado
Em Manaus há 17191 casos classificados como Em análise
Em Manaus há 53359 casos classificados como Descartado


### Conclusão

#### Em Manaus há cumulativamente 36671 casos confirmados, os casos em análise somam 17191 e um total de 53359 casos descartados.

#### 3. A qual período de tempo a base de dados se refere, isto é, qual o registro mais antigo e qual o mais recente? Leve em conta a data de notificação.

In [ ]:
import pandas as pd

df = pd.read_csv("C:/Users/hiago/git/Analise-COVID19-PP1-RNA2020.1/dataset/dataset_covid19_manaus.csv", encoding="ISO-8859-1", sep=";")
